graphs/uniform_search_cost.py

```diff
@@ -0,0 +1,260 @@
+
+        x = current[0]
+        y = current[1]
+        dist[x][y] = 0
+        final_cnt = 0
+        heap = [[0, x, y]]
+        while len(heap) > 0:
+            heap.sort(key=lambda x: x[0])
+            d, x, y = heap[0]
+            heap.pop(0)
+            if visited[x][y]:
+                continue
+            visited[x][y] = 1
+            if [x, y] in final:
+                idxs = [ix for ix, iy in enumerate(final) if iy == [x, y]]
+                if len(idxs) > 1:
+                    print("Twice")
+                if goal_answer[idxs[0]] == 10**8:
+                    final_cnt += 1
+                    print("Increment")
+                if goal_answer[idxs[0]] > d:
+                    goal_answer[idxs[0]] = d
+                    print("Extended distance")
+                if final_cnt == len(final):
+                    path = self.get_shortest_path(current, final[0], dist, dxy)
+                    return path
+            for dx, dy in dxy:
+                rx = dx + x
+                ry = dy + y
+                if 0 <= rx < self.m and 0 <= ry < self.n and grid[rx][ry] != 1:
+                    weight = 1
+                    # diagonal_weight = 1.414
+                    # if (
+                    #     (dx == -1 and dy == -1)
+                    #     or (dx == 1 and dy == 1)
+                    #     or (dx == 1 and dy == -1)
+                    #     or (dx == -1 and dy == 1)
+                    # ):
+                    #     new_dist = d + diagonal_weight
+                    # else:
+                    new_dist = d + weight
+                    if new_dist < dist[rx][ry]:
+                        dist[rx][ry] = new_dist
+                        heap.append([new_dist, rx, ry])
+        return None
+
+    def your_algorithm(
+        self, start_point: list[int], end_point: list[int], grid: list[list[int]]
+    ) -> list[list[float]] | list[list[int]] | None:
+        """
+        Return 2D list where optimal path is stored.
+        """
+        dxy = []
+        if start_point[1] - end_point[1] == 0 and start_point[0] - end_point[0] < 0:
+            dxy = dxy5
+        elif start_point[1] - end_point[1] == 0 and start_point[0] - end_point[0] > 0:
+            dxy = dxy6
+        elif start_point[0] - end_point[0] == 0 and start_point[1] - end_point[1] < 0:
+            dxy = dxy4
+        elif start_point[0] - end_point[0] == 0 and start_point[1] - end_point[1] > 0:
+            dxy = dxy3
+        elif start_point[0] - end_point[0] > 0:
+            dxy = dxy2
+        elif start_point[0] - end_point[0] < 0:
+            dxy = dxy1
+        goal_answer = []
+        for _ in range(0, len(end_point)):
+            goal_answer.append(10**8)
+        path = self.ucs(start_point, [end_point], grid, dxy, goal_answer)
+        if path is None:
+            return None
+        return path
+
+
+if __name__ == "__main__":
+    executed_object = UniformCostSearch(
```

Comment:
dont call the funciton here, use doctests. See other merged PRs.

Reply:
Where should best place for the function call@rohan472000 ?

Reply:
Kindly see other merged PR or existing .py files in repo to see how to use doctest.

Reply:
Thanks @rohan472000, learned a lot on doctest. Just changed my code and also added learning source.